io.c

```diff
@@ -10122,142 +10145,15 @@ rb_io_advise(int argc, VALUE *argv, VALUE io)
- *  The record consists of multiple bytes.
- *  So, the remote side of SSL sends a partial record, IO.select
- *  notifies readability but OpenSSL::SSL::SSLSocket cannot decrypt a
- *  byte and OpenSSL::SSL::SSLSocket#readpartial will block.
- *
- *  Also, the remote side can request SSL renegotiation which forces
- *  the local SSL engine to write some data.
- *  This means OpenSSL::SSL::SSLSocket#readpartial may invoke #write
- *  system call and it can block.
- *  In such a situation, OpenSSL::SSL::SSLSocket#read_nonblock raises
- *  IO::WaitWritable instead of blocking.
- *  So, the caller should wait for ready for writability as above
- *  example.
- *
- *  The combination of nonblocking methods and IO.select is also useful
- *  for streams such as tty, pipe socket socket when multiple processes
- *  read from a stream.
- *
- *  Finally, Linux kernel developers don't guarantee that
- *  readability of select(2) means readability of following read(2) even
- *  for a single process.
- *  See select(2) manual on GNU/Linux system.
- *
- *  Invoking IO.select before IO#readpartial works well as usual.
- *  However it is not the best way to use IO.select.
- *
- *  The writability notified by select(2) doesn't show
- *  how many bytes are writable.
- *  IO#write method blocks until given whole string is written.
- *  So, <code>IO#write(two or more bytes)</code> can block after
- *  writability is notified by IO.select.  IO#write_nonblock is required
- *  to avoid the blocking.
- *
- *  Blocking write (#write) can be emulated using #write_nonblock and
- *  IO.select as follows: IO::WaitReadable should also be rescued for
- *  SSL renegotiation in OpenSSL::SSL::SSLSocket.
- *
- *    while 0 < string.bytesize
- *      begin
- *        written = io_like.write_nonblock(string)
- *      rescue IO::WaitReadable
- *        IO.select([io_like])
- *        retry
- *      rescue IO::WaitWritable
- *        IO.select(nil, [io_like])
- *        retry
- *      end
- *      string = string.byteslice(written..-1)
- *    end
- *
- *  === Parameters
- *  read_array:: an array of IO objects that wait until ready for read
- *  write_array:: an array of IO objects that wait until ready for write
- *  error_array:: an array of IO objects that wait for exceptions
- *  timeout:: a numeric value in second
- *
- *  === Example
- *
- *      rp, wp = IO.pipe
- *      mesg = "ping "
- *      100.times {
- *        # IO.select follows IO#read.  Not the best way to use IO.select.
- *        rs, ws, = IO.select([rp], [wp])
- *        if r = rs[0]
- *          ret = r.read(5)
- *          print ret
- *          case ret
- *          when /ping/
- *            mesg = "pong\n"
- *          when /pong/
- *            mesg = "ping "
- *          end
- *        end
- *        if w = ws[0]
- *          w.write(mesg)
- *        end
- *      }
- *
- *  <em>produces:</em>
- *
- *      ping pong
- *      ping pong
- *      ping pong
- *      (snipped)
- *      ping
```

Comment:
@BurdetteLamar Either I haven't noticed it previously, or is it a recent change?
Why have you removed most of the docs of the method? 
While it might be said that they were unusually long, they were quite clear, demonstrated interesting examples and described important quirks. And now they are just gone?.. What's the idea behind this?